### DISCUSSION TITLE
[Feature]: Optionally move deleted items to a recycle bin

### DISCUSSION THREAD

**Discussion by JVT038** [immich-app/immich#1652] [Feature Request]
### Feature detail

If a user deletes an item, perhaps it was done on purpose, or maybe the user needs the item again a week later.
For this, create a recycle bin, where the items are temporarily stored for an amount of time specified by the user.
If the user decides they want to recover an item, they can just go to some other tab, look for their deleted item, and recover it. 
Also, if this feature is going to be implemented, make it optional. If the admin decides they don't want this and instantly delete items when they're deleted, they can just turn this off.

I selected web as platform, but it would be nice for the phone as well, and there are probably some server-side changes necessary for this.

### Platform

Web

**Comment by Dilanwizzy** [comment]
I've just been thinking about this feature and was going to open up an issue to implement this. I want to start contributing to the repo, and I will pick this up.

**Reply by lorvent** [reply]
just wondering, if you started working on this....

**Comment by mouie** [comment]
+1 this would be a great feature

I'd envision it being similar to the Archive functionality in terms of build and with an added auto empty after X time period. This will help reverse any mistaken deletes given it is not unreasonable to fat finger the mobile app screen and delete a photo that is only hosted on the server.

**Reply by AngelaDMerkel** [reply]
This is the biggest incentive to introduce this feature. Accidents can happen and it would be in line with the goals of a photo backup service to protect against unintentional loss, even if caused by a user action. The additional storage space involved would be negligible if there is a delete after X time function.

**Reply by mouie** [reply]
Awesome to see this added to v1.82.0
